RiotSwiftUI/Modules/Room/Composer/Test/Unit/ComposerViewModelTests.swift

```diff
@@ -0,0 +1,68 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@testable import RiotSwiftUI
+import SwiftUI
+import XCTest
+
+class ComposerViewModelTests: XCTestCase {
+    var viewModel: ComposerViewModel!
+    var context: ComposerViewModel.Context!
+    
+    override func setUpWithError() throws {
+        viewModel = ComposerViewModel(initialViewState: ComposerViewState())
+        context = viewModel.context
+    }
+    
+    func testSendState() {
+        viewModel.sendMode = .send
+        XCTAssert(context.viewState.sendMode == .send)
+        XCTAssert(context.viewState.shouldDisplayContext == false)
+        XCTAssert(context.viewState.eventSenderDisplayName == nil)
+        XCTAssert(context.viewState.contextImageName == nil)
+        XCTAssert(context.viewState.contextDescription == nil)
+    }
+    
+    func testEditState() {
+        viewModel.sendMode = .edit
+        XCTAssert(context.viewState.sendMode == .edit)
+        XCTAssert(context.viewState.shouldDisplayContext == true)
+        XCTAssert(context.viewState.eventSenderDisplayName == nil)
+        XCTAssert(context.viewState.contextImageName == Asset.Images.inputEditIcon.name)
+        // Need to improve this test
+        XCTAssert(context.viewState.contextDescription != nil)
```

Comment:
@langleyd how do we test efficiently for localised texts like in this case? Do we assume english version as the test version always? Or do I need to use the VectorL10 module again?

Reply:
okay will add these real quickly, and then merge it if the lest looks good to you!